### ISSUE TITLE
Broken File with Red Bar Error of Unknown Cause

### ISSUE THREAD

**Issue by Gerstmann-Bradley** [JacquesLucke/animation_nodes#1838]
People shared a file to me. For some reason I saw red bar error that's not possible to be fixed.
On his side, it seems to have no red bar error but other problem instead.
The issue does not present to me if everything is built from scratch.
![image](https://user-images.githubusercontent.com/55994236/160232712-5b4e18ad-771c-4c2b-9ae1-0a0a1911ce06.png)
[Shared File.zip](https://github.com/JacquesLucke/animation_nodes/files/8355044/Shared.File.zip)

**Comment by 3DSinghVFX** [comment]
@Gerstmann-Bradley It is working fine on my side.

**Comment by OmarEmaraDev** [comment]
Seems to work here as well. Notice that there are some removed links in the latest version by the way:
```
WARNING: link removed
    Tree: 'Preset1'
    From Socket: 'Vector Noise' -> 'values'
    To Socket: 'Combine Vector.009' -> 'x'

WARNING: link removed
    Tree: 'Preset1'
    From Socket: 'Vector Noise.001' -> 'values'
    To Socket: 'Combine Vector.009' -> 'y'

WARNING: link removed
    Tree: 'Preset1'
    From Socket: 'Vector Noise.002' -> 'values'
    To Socket: 'Combine Vector.009' -> 'z'
```

**Comment by Gerstmann-Bradley** [comment]
I am using Blender 3.1 and "animation_nodes_v2_3_windows_py310" from AN+EN.
here are the console result:

Traceback (most recent call last):
  File "C:\Users\******\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\animation_nodes\execution\units.py", line 57, in setupExecutionUnits
    executionUnits = getExecutionUnits(nodeTrees)
  File "C:\Users\******\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\animation_nodes\execution\units.py", line 103, in getExecutionUnits
    for subprogramID in getNeededSubprogramIDs(network):
  File "C:\Users\******\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\animation_nodes\execution\units.py", line 111, in getNeededSubprogramIDs
    subprogramIDs.update(getNeededSubprogramIDs(subNetwork))
  File "C:\Users\******\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\animation_nodes\execution\units.py", line 108, in getNeededSubprogramIDs
    subprogramIDs = network.referencedSubprogramIDs.copy()
AttributeError: 'NoneType' object has no attribute 'referencedSubprogramIDs'
Traceback (most recent call last):
  File "C:\Users\******\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\animation_nodes\utils\handlers.py", line 85, in depsgraphUpdatePost
    handler(scene, depsgraph)
  File "C:\Users\******\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\animation_nodes\events.py", line 46, in sceneChanged
    event_handler.update(event.getActives().union({"Scene"}))
  File "C:\Users\******\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\animation_nodes\utils\recursion.py", line 12, in wrapper
    result = function(*args, **kwargs)
  File "C:\Users\******\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\animation_nodes\event_handler.py", line 25, in update
    executeNodeTrees(nodeTrees)
  File "C:\Users\******\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\animation_nodes\execution\auto_execution.py", line 13, in executeNodeTrees
    nodeTree.autoExecute()
  File "C:\Users\******\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\animation_nodes\base_types\node_tree.py", line 84, in autoExecute
    self._execute()
  File "C:\Users\******\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\animation_nodes\base_types\node_tree.py", line 102, in _execute
    success = unit.execute()
  File "C:\Users\******\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\animation_nodes\execution\main_execution_unit.py", line 73, in raiseNotSetupException
    raise ExecutionUnitNotSetup()
animation_nodes.problems.ExecutionUnitNotSetup
Info: Read library:  'C:\Users\******\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\Bradley's Geo Node Presets\Data\3.1\preset.blend', 'C:\Users\******\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\Bradley's Geo Node Presets\Data\3.1\preset.blend', parent '<direct>'

@OmarEmaraDev Yes, that was the "other problem" people encountered.
he reported that everything works well with these group nodes saved in his startup.
but once he close blender and open the file again, group nodes don't work because of this.
Do you have any idea about that?

**Comment by OmarEmaraDev** [comment]
This was fixed this week in 1153368. So maybe the AN version is not up to date?

**Comment by Gerstmann-Bradley** [comment]
> This was fixed this week in [1153368](https://github.com/JacquesLucke/animation_nodes/commit/1153368658c5bf63c32abfcd731399e0c6b27f2b). So maybe the AN version is not up to date?

My bad, didn't realize this fix. I've updated my AN and it resolves the red bar error.
But I can't reproduce the "lost link" thing reported by the owner of this file.
Do you have any idea?
Also, perhaps it's irrelevant, do you know why when it's "Scene Change" on, the AN keeps updating itself in that file?
I don't see anything that seems to change the scene in my opinion?

**Comment by 3DSinghVFX** [comment]
> > This was fixed this week in [1153368](https://github.com/JacquesLucke/animation_nodes/commit/1153368658c5bf63c32abfcd731399e0c6b27f2b). So maybe the AN version is not up to date?
> 
> My bad, didn't realize this fix. I've updated my AN and it resolves the red bar error. But I can't reproduce the "lost link" thing reported by the owner of this file. Do you have any idea? 

That is due Vector Noise node, in AN+EN we have also the Blender Noises so if you open the file of AN+EN with AN then it may break the Vector Noise node connections.

**Comment by Gerstmann-Bradley** [comment]
> That is due Vector Noise node, in AN+EN we have also the Blender Noises so if you open the file of AN+EN with AN then it may break the Vector Noise node connections.

Thanks for reminder, I asked the owner to use AN+EN it seems to solve the problem.
I guess it's because the subprogram was originally made in AN+EN, and he opened with AN which always causing his problem of link remove, while I don't have any problem whatsoever.
Thanks for the help.
